brave/brave-ios
# Fix #7360: Add non-aggressive cosmetic filters

## Summary of Changes


This pull request fixes #7360

This adds code to support standard mode but until UI is available, we cannot enable it. For this ticket we are testing for unchanged (aggressive) cosmetic filters. Standard mode can only be enabled through code changes for the time being and will be tested once the UI is ready. The UI is done in the following issue: #7352 

Note: The logs will be removed before merging. Feel free to leave a comment but it will be the last thing I will do because it is difficult to test scripts in webkit.

## Submitter Checklist:


## Test Plan:

NOTE: These tests are also covered [here](https://github.com/brave/brave-ios/pull/7437) which will be deployed together with this PR. You can skip these tests here as they duplicated there.

1. Test any known cosmetic filtering for pages like reddit:
2. Go to brave search and search for "trezor". 
- [ ] The ad on the top of the page should be shown on standard mode and hidden on aggressive.
3. Enable "Block cookie consent notices" and go to reddit. 
- [ ] Cookie notices should be blocked on standard and aggressive
4. Enable "Block open in app notices" and go to reddit. 
- [] Open in app notices should be blocked on standard and aggressive
5. Go to the following page: https://dev-pages.brave.software/filtering/cosmetic-filtering.html
- [ ] On aggressive mode, all images should be hidden after clicking "Run tests"
- [ ] On standard mode, only 3rd party images should be hidden after clicking "Run tests"

Note: Without the UI [here](https://github.com/brave/brave-ios/pull/7437) the script is set to standard mode. You cannot test aggressive mode without this code merged.

## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
